MicroPython/src/controller/main_controller.py

```diff
@@ -65,97 +76,145 @@ def perform(self):
-            self.last_test_result = result[0]
-
-            self.is_test_mode_running = False
-            self.send_callback(event, True)
+        elif event.event_type == MainControllerEventType.TEST_CONNECTION:
+            logging.debug("WAITING FOR WIFI CONFIG")
+            while not config.cfg.ap_config_done:
+                pass
+            logging.debug("GOT WIFI CONFIG")
+            logging.debug("Testing AWS connection")
+
+            MainController.test_connection_with_wifi_and_cloud()
+
+            config.cfg.tested_connection_cloud = True
+            config.cfg.save()
+
+        elif event.event_type == MainControllerEventType.PRINT_TIME:
+            logging.debug("WAITING FOR TESTED CONNECTION CLOUD")
+            while not config.cfg.tested_connection_cloud:
+                pass
+            logging.debug("GOT TESTED CONNECTION CLOUD")
+
+            self.print_time()
+
+            config.cfg.printed_time = True
+            config.cfg.save()
+
+        elif event.event_type == MainControllerEventType.GET_SENSOR_DATA:
+            logging.debug("WAITING FOR PRINTED TIME")
+            while not config.cfg.printed_time:
+                pass
+            logging.debug("GOT PRINTED TIME")
+
+            logging.debug("GETTING SENSOR DATA")
+
+            self.data_acquisitor.acquire_temp_humi()
+
+            config.cfg.got_sensor_data = True
+            config.cfg.save()
+
+        elif event.event_type == MainControllerEventType.PUBLISH_DATA:
+            logging.debug("WAITING FOR GOT SENSOR DATA")
+            while not config.cfg.got_sensor_data:
+                pass
+            logging.debug("GOT GOT SENSOR DATA")
+            logging.debug("Publishing data to cloud")
+
+            self.publish_data()
+
+            config.cfg.published_to_cloud = True
+            config.cfg.save()
+
+        elif event.event_type == MainControllerEventType.GO_TO_SLEEP:
+            logging.debug("WAITING FOR PUBLISHED TO CLOUD")
+            while not config.cfg.published_to_cloud:
+                pass
+            logging.debug("GOT PUBLISHED TO CLOUD")
+
+            self.go_to_sleep(event)
 
         elif event.event_type == MainControllerEventType.ERROR_OCCURRED:
             logging.debug("Processing event error occurred")
             print(event.data)
-            self.event_handler(event.data)
             self.send_callback(event, True)
 
         else:
             quit(-1)
 
-    def event_handler(self, *event_data):
-        pass
-
     @staticmethod
-    def send_callback(event, data):
+    def send_callback(event: MainControllerEvent, data: object) -> None:
+        """
+        Executing event's callback function.
+        :param event: Event's object.
+        :param data: Callback function parameters.
+        :return: None
+        """
         if event.callback:
             event.callback(data)
 
-    def get_measurement(self):
+    def get_measurement(self) -> int:
```

Comment:
Is this method necessary? I see, that webapp call this to show single measurement for user, but I thinks this not works properly. Could you delete whole funcionality? Including webapp if you can?